Detectors/TRD/base/src/TRDCommonParam.cxx

```diff
@@ -163,91 +165,71 @@ float TRDCommonParam::GetOmegaTau(float vdrift)
 //_____________________________________________________________________________
 bool TRDCommonParam::GetDiffCoeff(float& dl, float& dt, float vdrift)
 {
-  /*
   //
   // Calculates the diffusion coefficients in longitudinal <dl> and
   // transverse <dt> direction for a given drift velocity <vdrift>
   //
 
   // Nothing to do
-  if (TMath::Abs(fDiffLastVdrift - vdrift) < 1.e-3) {
-
-    dl = fDiffusionL;
-    dt = fDiffusionT;
+  if (TMath::Abs(mDiffLastVdrift - vdrift) < 0.001) {
+    dl = mDiffusionL;
+    dt = mDiffusionT;
     return true;
-
   }
 
-  fDiffLastVdrift = vdrift;
-
-  if      (IsXenon()) {
-
+  mDiffLastVdrift = vdrift;
+  if (IsXenon()) {
     //
     // Vd and B-field dependent diffusion and Lorentz angle
     //
 
     // The magnetic field strength
-    AliMagF* fld = (AliMagF *) TGeoGlobalMagField::Instance()->GetField();
+    o2::field::MagneticField* fld = static_cast<o2::field::MagneticField*>(TGeoGlobalMagField::Instance()->GetField());
     if (!fld) {
+      LOG(FATAL) << "Magnetic field is not initialized!";
       return false;
     }
-    double field = 0.1 * fld->SolenoidField();   // kGauss -> Tesla
+    double field = 0.1 * fld->solenoidField(); // kGauss -> Tesla
 
     // DiffusionL
     const int kNbL = 5;
-    float p0L[kNbL] = {  0.007440,  0.007493,  0.007513,  0.007672,  0.007831 };
-    float p1L[kNbL] = {  0.019252,  0.018912,  0.018636,  0.018012,  0.017343 };
-    float p2L[kNbL] = { -0.005042, -0.004926, -0.004867, -0.004650, -0.004424 };
-    float p3L[kNbL] = {  0.000195,  0.000189,  0.000195,  0.000182,  0.000169 };
+    const float p0L[kNbL] = { 0.007440, 0.007493, 0.007513, 0.007672, 0.007831 };
+    const float p1L[kNbL] = { 0.019252, 0.018912, 0.018636, 0.018012, 0.017343 };
+    const float p2L[kNbL] = { -0.005042, -0.004926, -0.004867, -0.004650, -0.004424 };
+    const float p3L[kNbL] = { 0.000195, 0.000189, 0.000195, 0.000182, 0.000169 };
 
-    int ibL = ((int) (10 * (field - 0.15)));
-    ibL       = TMath::Max(     0,ibL);
-    ibL       = TMath::Min(kNbL-1,ibL);
+    int ibL = ((int)(10 * (field - 0.15)));
+    ibL = TMath::Max(0, ibL);
+    ibL = TMath::Min(kNbL - 1, ibL);
 
```

Comment:
The ibL and ibT can be precalculated as data members in a single initialization call. 
Also, I would precalculate ``float v2=vdrift*vrdrift, v3=v2*vdrift;`` and use them in the mDiffusionL and mDiffusionT calculations below.
The same applies to GetOmegaTau method. Eventually, we should substitute slow functions like ``tan()`` by less precise and faster methods wheneve the high precision is not needed (like in GetOmegaTau), but this can be done later